src/pubsub/mod_pubsub_db_rdbms.erl

```diff
@@ -86,6 +83,146 @@
+    mongoose_rdbms:prepare(pubsub_delete_item, pubsub_items,
+        [nidx, created_luser, created_lserver, itemid],
+        <<"DELETE FROM pubsub_items WHERE nidx = ? AND created_luser = ? ",
+          "AND created_lserver = ? AND itemid = ?">>),
+    mongoose_rdbms:prepare(pubsub_delete_all_items, pubsub_items, [nidx],
+        <<"DELETE FROM pubsub_items WHERE nidx = ?">>),
+    ItemColumns = item_columns(),
+    mongoose_rdbms:prepare(pubsub_get_item, pubsub_items, [nidx, itemid],
+        <<"SELECT ", ItemColumns/binary, " FROM pubsub_items WHERE nidx = ? AND itemid = ?">>),
+    mongoose_rdbms:prepare(pubsub_del_item, pubsub_items, [nidx, itemid],
+        <<"DELETE FROM pubsub_items WHERE nidx = ? AND itemid = ?">>),
+    mongoose_rdbms:prepare(pubsub_del_items, pubsub_items, [nidx, itemids],
+        <<"DELETE FROM pubsub_items WHERE nidx = ? AND itemid IN (?)">>),
+
+    % ------------------- Nodes --------------------------------
+    mongoose_rdbms:prepare(pubsub_update_pubsub_node, pubsub_nodes, [type, owners, options, nidx],
+        <<"UPDATE pubsub_nodes SET type = ?, owners = ?, options = ? WHERE nidx = ?">>),
+    PubsubNodeFields = pubsub_node_fields(),
+    mongoose_rdbms:prepare(pubsub_select_node_by_key_and_name, pubsub_nodes, [key, name],
+        <<"SELECT ", PubsubNodeFields/binary, " from pubsub_nodes WHERE p_key = ? AND name = ?">>),
+    mongoose_rdbms:prepare(pubsub_select_node_by_id, pubsub_nodes, [nidx],
+        <<"SELECT ", PubsubNodeFields/binary, " from pubsub_nodes WHERE nidx = ?">>),
+    mongoose_rdbms:prepare(pubsub_select_nodes_by_key, pubsub_nodes, [key],
+        <<"SELECT ", PubsubNodeFields/binary, " from pubsub_nodes WHERE p_key = ?">>),
+    mongoose_rdbms:prepare(pubsub_select_nodes_in_list_with_key, pubsub_nodes, [key, names],
+        <<"SELECT ", PubsubNodeFields/binary, " from pubsub_nodes "
+          "WHERE p_key = ? AND name IN (?)">>),
+    PubsubNodeFieldsPrefixed = pubsub_node_fields("pn"),
+    mongoose_rdbms:prepare(pubsub_select_nodes_by_affiliated_user,
+        pubsub_affiliations, [luser, lserver],
+        <<"SELECT aff, ", PubsubNodeFieldsPrefixed/binary, " FROM pubsub_affiliations AS pa "
+          "INNER JOIN pubsub_nodes AS pn ON pa.nidx = pn.nidx WHERE luser = ? AND lserver = ?">>),
+    mongoose_rdbms:prepare(pubsub_select_subnodes_empty_node, pubsub_nodes, [key],
+        <<"SELECT ", PubsubNodeFieldsPrefixed/binary, " from pubsub_nodes as pn "
+          "LEFT JOIN pubsub_node_collections as collection ON pn.name = collection.name "
+          "WHERE p_key = ? AND collection.parent_name IS NULL">>),
+    mongoose_rdbms:prepare(pubsub_select_subnodes, pubsub_nodes, [parent_name, key],
+        <<"SELECT ", PubsubNodeFieldsPrefixed/binary, " from pubsub_nodes as pn "
+          "INNER JOIN pubsub_node_collections as collection ON pn.name = collection.name AND "
+          "collection.parent_name = ? WHERE p_key = ?">>),
```

Comment:
This query whole not work, because it has arguments from two tables.
I would have to add that feature.